src/components/viz/Timeline/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import styles from './styles.scss';
+
+export type MinMax = [number, number];
+
+export interface TimelineProps<T> {
+    className?: string;
+    data: T[];
+    keySelector: (item: T) => string | number;
+    valueSelector: (item: T) => number;
+    labelSelector: (item: T) => React.ReactNode;
+    tickLabelSelector: (item: number) => React.ReactNode;
+    domain?: MinMax;
+}
+
+
+const scale = (domain: MinMax, range: MinMax, value: number) => {
+    const [minDomain, maxDomain] = domain;
+    const [minRange, maxRange] = range;
+    const normalizedValue = (value - minDomain) / (maxDomain - minDomain);
+    const newValue = (normalizedValue * (maxRange - minRange)) + minRange;
+    return newValue;
+};
+
+function Timeline<T>(props: TimelineProps<T>) {
+    const {
+        className,
+        labelSelector,
+        valueSelector,
+        keySelector,
+        tickLabelSelector,
+        data,
+        domain: domainFromProps,
+    } = props;
+
+    const domain: MinMax = React.useMemo(() => domainFromProps ?? [
+        Math.min(...data.map(valueSelector)),
+        Math.max(...data.map(valueSelector)),
```

Comment:
We should verify the length of data here.
Math.min() is Infinity and Math.max() is -Infinity.